utilities/imgcat

```diff
@@ -118,40 +159,67 @@ while [ $# -gt 0 ]; do
     -p | --p | --print)
         print_filename=1
         ;;
+    -q | --q | --quiet)
+        print_filename=0
+        ;;
+    -W | --W | --width)
+        validate_size_unit "$2"
+        width="$2"
+        shift
+        ;;
+    -H | --H | --height)
+        validate_size_unit "$2"
+        height="$2"
+        shift
+        ;;
+    -r | --R | --preserve-aspect-ratio)
+        preserve_aspect_ratio=1
+        ;;
+    -s | --S | --stretch)
+        preserve_aspect_ratio=0
+        ;;
+    -f | --F | --file)
+        has_stdin=f
+        is_url=f
+        ;;
     -u | --u | --url)
         check_dependency curl
-        encoded_image=$(curl -s "$2" | b64_encode) || (
-            error "No such file or url $2"
-            exit 2
-        )
         has_stdin=f
-        print_image "$2" 1 "$encoded_image" "$print_filename"
-        set -- "${@:1:1}" "-u" "${@:3}"
-        if [ "$#" -eq 2 ]; then
-            exit
-        fi
+        is_url=t
         ;;
     -*)
         error "Unknown option flag: $1"
         show_help
         exit 1
         ;;
     *)
-        if [ -r "$1" ]; then
-            has_stdin=f
-            print_image "$1" 1 "$(b64_encode <"$1")" "$print_filename"
+        if [ "$is_url" == "t" ]; then
+            encoded_image=$(curl -fs "$1" | b64_encode) || {
+                error "Could not retrieve image from URL $1, error_code: $?"
```

Comment:
`errcho`

Reply:
We're should not confuse them, because `errcho()` is about output to STDERR and `error()` is for printing an `ERROR:` message. This is convenient to separate them in case if we need to add option to suppress printing of all error messages.

Of course we can reuse `errcho()` in `error()`.